frontend/src/bundles/common/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,33 @@
+import { Loader, Navigate } from '~/bundles/common/components/components.js';
+import { AppRoute, DataStatus } from '~/bundles/common/enums/enums.js';
+import {
+    useAppSelector,
+    useEffect,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+
+interface Properties {
+    children: React.ReactNode;
+}
+
+const ProtectedRoute: React.FC<Properties> = ({ children }) => {
+    const [isLoading, setIsLoading] = useState(true);
```

Comment:
To manage the component's loading state, since when the page loads, the user is null, and without this state, it would redirect us to the sign-in page. With this state, we can ensure that the loader will be shown until the request is completed or the user is actually not found.